userdocs/diagnostics/sending-closure-risks-data-race.md

```diff
@@ -0,0 +1,65 @@
+# Sending closure risks causing data races
+
+If a type does not conform to `Sendable` the compiler will enforce that each instance of that type is only accessed by one concurrency domain at a time. The compiler will also prevent you from capturing values in closures that are sent to another concurrency domain if the value can be accessed from the original concurrency domain too.
+
+For example:
+
+```swift
+class MyModel {
+  func perform() {
+    Task {
+      self.update()
+    }
+  }
+
+  func update() { ... }
+}
+```
+
+Building with complete concurrency checking will diagnose the capture of `self` in the task closure:
+
+```
+| class MyModel {
+|   func perform() {
+|     Task {
+|     `- error: passing closure as a 'sending' parameter risks causing data races between code in the current task and concurrent execution of the closure
+|       self.update()
+|            `- note: closure captures 'self' which is accessible to code in the current task
+|     }
+|   }
+```
+
+This code is invalid because the task that calls `perform()` runs concurrently with the task that calls `update()`. The `MyModel` type does not conform to `Sendable`, so this means the concurrent tasks can access mutable state simultaneously.
```

Comment:
```suggestion
This code is invalid because the task that calls `perform()` runs concurrently with the task that calls `update()`. The `MyModel` type isn't marked as conforming to `Sendable`, which tells the compiler that the concurrent tasks could access mutable state simultaneously.
```